internal/driver/controllerserver_helper.go

```diff
@@ -403,25 +431,38 @@ func (cs *ControllerServer) validateCreateVolumeRequest(ctx context.Context, req
 // prepareVolumeParams prepares the volume parameters for creation.
 // It extracts the capacity range from the request, calculates the size,
 // and generates a normalized volume name. Returns the volume name and size in GB.
-func (cs *ControllerServer) prepareVolumeParams(ctx context.Context, req *csi.CreateVolumeRequest) (volumeName string, targetSizeGB int, size int64, err error) {
+func (cs *ControllerServer) prepareVolumeParams(ctx context.Context, req *csi.CreateVolumeRequest) (volumeName string, targetSizeGB int, size int64, encryptionStatus string, err error) {
 	log := logger.GetLogger(ctx)
 	log.V(4).Info("Entering prepareVolumeParams()", "req", req)
 	defer log.V(4).Info("Exiting prepareVolumeParams()")
 
+	// Check encryption status and if it's supported in the specified region.
+	encryptionStatus = "disabled"
 	// Retrieve the capacity range from the request to determine the size limits for the volume.
 	capRange := req.GetCapacityRange()
 	// Get the requested size in bytes, handling any potential errors.
 	size, err = getRequestCapacitySize(capRange)
 	if err != nil {
-		return "", 0, 0, err
+		return "", 0, 0, encryptionStatus, err
 	}
 
 	preKey := linodevolumes.CreateLinodeVolumeKey(0, req.GetName())
 	volumeName = preKey.GetNormalizedLabelWithPrefix(cs.driver.volumeLabelPrefix)
 	targetSizeGB = bytesToGB(size)
+	// Check if encryption should be enabled
+	if req.GetParameters()[VolumeEncryption] == True {
+		supported, err := cs.isEncryptionSupported(ctx, cs.metadata.Region)
+		if err != nil {
+			return "", targetSizeGB, size, encryptionStatus, nil
+		}
+		if !supported {
+			return "", targetSizeGB, 0, encryptionStatus, errInternal("Volume encryption is not supported in the %s region", cs.metadata.Region)
```

Comment:
question: any reason for not returning volume name & size here?

Reply:
No reason. I think I missed it.